compiler/src/iree/compiler/API2/test/CMakeLists.txt

```
@@ -33,4 +33,16 @@ if(TARGET iree_compiler_API2_test_api-test-binary)
     OUTPUT_NAME "test-iree-compiler-api-test-binary"
     RUNTIME_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}/tools"
   )
+
+  if(BUILD_SHARED_LIBS)
+    # When the compiler/LLVM is built in shared library mode, we can't
+    # do C++ linking against re-exported libraries, so we have to link
+    # against the parts that the test uses of the MLIR API explicitly.
```

Nit: word order here had me re-reading a couple times to parse

```suggestion
    # against the parts of the MLIR API that the test uses explicitly.
```